crates/astria-core/src/primitive/v1/asset.rs

```diff
@@ -83,6 +88,30 @@ impl Denom {
     pub fn to_base_denom(&self) -> Self {
         Self::from_base_denom(&self.base_denom)
     }
+
+    /// Create a new [`Denom`] with the given prefix removed.
+    ///
+    /// # Errors
+    ///
+    /// - if the denom does not have the given prefix.
+    pub fn remove_prefix(&self, prefix: &str) -> Result<Self, InvalidPrefixToRemove> {
+        let denom_trace = self
+            .to_string()
```

Comment:
Why allocate a new string prior to removing `strip_prefix`? Because one sometimes wants to remove a prefix that reaches "into" the `denom` part?

Nomenclature wise this leaves me pretty confused. Just looking at the definition of `Denom` I see a `prefix` and a `denom`. So why would a method called `remove_prefix` also remove parts of the `Denom`?

Reply:
the `Denom` consists of a prefix and a base denom (I'll rename the `denom` field to `base_denom`), this is just a "denomination trace" broken down into parts (eg. `transfer/channel-1/utia` is broken down into `transfer/channel-1` and `utia`). so removing a prefix is removing some prefix from the entire denomination trace. is there a rename I could do to make this clearer?